kaitaistruct.py

```diff
@@ -15,6 +15,18 @@
 #
 __version__ = '0.8'
 
+from RecursionSafe import RecursionSafe
+
+recSafe=RecursionSafe()
+
+@recSafe.wrap
+def repr_generator_for_all_props(self):
+  """Generator to use in own __repr__ functions."""
+  return (
+    "".join(( str(k), "=", repr(getattr(self, k)) ))
+    for k in dir(self)
+    if k[0] != "_" and not hasattr(KaitaiStruct, k) and not isinstance(getattr(self, k), type)
```

Comment:
`k.startswith('_')` would be more expressive.

Reply:
It's longer and I'm not sure what is more performant.